src/game/logic/object/locomotor.cpp

```diff
@@ -1853,3 +1908,295 @@ void Locomotor::Start_Move()
+            force3d.x = force * dir->x;
+            force3d.y = force * dir->y;
+            force3d.z = 0.0f;
+            physics->Apply_Motive_Force(&force3d);
+        }
+    }
+}
+
+void Locomotor::Maintain_Current_Position_Wings(Object *obj, PhysicsBehavior *physics)
+{
+    if (!Get_Flag(MAINTAIN_POS_IS_VALID)) {
+        captainslog_dbgassert(false, "invalid maintain pos");
+    }
+
+    physics->Set_Turning(TURN_NONE);
+
+    if (physics->Is_Motive() && obj->Is_Above_Terrain()) {
+        BodyDamageType damage = obj->Get_Body_Module()->Get_Damage_State();
+        float radius = m_template->m_circlingRadius;
+
+        if (radius == 0.0f) {
+            radius = Calc_Min_Turn_Radius(damage, nullptr);
+        }
+
+        float x = m_maintainPos.x - obj->Get_Position()->x;
+        float y = m_maintainPos.y - obj->Get_Position()->y;
+        float orientation;
+
+        if (Within_Epsilon(x) && Within_Epsilon(y)) {
+            orientation = obj->Get_Orientation();
+        } else {
+            orientation = GameMath::Atan2(y, x);
+        }
+
+        float angle = DEG_TO_RADF(157.5f);
+
+        if (radius < 0.0f) {
+            radius = -radius;
+            angle = -angle;
+        }
+
+        angle = orientation + angle;
+        Coord3D goal_pos = m_maintainPos;
+        goal_pos.x = GameMath::Cos(angle) * radius + goal_pos.x;
+        goal_pos.y = GameMath::Sin(angle) * radius + goal_pos.y;
+        Move_Towards_Position_Wings(obj, physics, goal_pos, 0.0f, m_template->m_minSpeed);
+    }
+}
+
+LocomotorSet::LocomotorSet() : m_validLocomotorSurfaces(0), m_downhillOnly(false) {}
+
+LocomotorSet::LocomotorSet(const LocomotorSet &that)
+{
+    captainslog_dbgassert(false, "unimplemented");
+}
+
+LocomotorSet::~LocomotorSet()
+{
+    Clear();
+}
+
+void LocomotorSet::Xfer_Snapshot(Xfer *xfer)
+{
+    uint8_t version = 1;
+    xfer->xferVersion(&version, 1);
+    uint16_t size = (uint16_t)m_locomotors.size();
+    xfer->xferUnsignedShort(&size);
+
+    if (xfer->Get_Mode() == XFER_SAVE) {
+        for (auto i = m_locomotors.begin(); i != m_locomotors.end(); i++) {
+            Locomotor *l = *i;
+            Utf8String name = l->Get_Template_Name();
+            xfer->xferAsciiString(&name);
+            xfer->xferSnapshot(l);
+        }
+    } else if (xfer->Get_Mode() == XFER_LOAD) {
+        if (!m_locomotors.empty()) {
+            captainslog_error("LocomotorSet::xfer - vector is not empty, but should be");
+            throw XFER_STATUS_NOT_EMPTY;
+        }
+
+        for (uint16_t j = 0; j < size; j++) {
+            Utf8String name;
+            xfer->xferAsciiString(&name);
+            LocomotorTemplate *lt = g_theLocomotorStore->Find_Locomotor_Template(Name_To_Key(name));
+
+            if (!lt) {
+                captainslog_debug("LocomotorSet::xfer - template %s not found", name.Str());
+                throw XFER_STATUS_NOT_FOUND;
+            }
+
+            Locomotor *l = new Locomotor(lt);
```

Comment:
this should be calling New_Locomotor of the loco store

Reply:
fixed.